src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -94,6 +94,14 @@ internal partial class Binder
 
                 foreach (var p in parameterSyntaxList.Value)
                 {
+                    if (p.AttributeLists.Count > 0 && syntax.Kind() == SyntaxKind.AnonymousMethodExpression)
```

Comment:
> So, to avoid the issue entirely, it's useful to just get rid of contextual parsing as much as possible in the parser.

I am not objecting to the change. However, it feels strange to me that the check here is specific only to AnonymousMethodExpression. I suggest to do it for lambdas as well, otherwise the logic would lead people to believe that attributes are allowed on lambda parameters.